### PR TITLE
Interface/stp

### LINKED ISSUE #905: Connect interface to current STP model to let us see roles
We currently can't see the robots' roles.
In widget.h there is a function getRoleForID() and if we modify that to use our current system everyhting should work.
Currently we have a map from roleName to ID, called STPInfos

### PR DESCRIPTION
**Pull request by Yuhanun**
Updated the interface to show STP properly.

STP for all:
image

STP for keeper:
image

Also added the role names below the robot names (it'll be below BS etc)

### CODE REVIEW

**Comment by RobotJente** [comment]
It looks nice! However, the scroll bar for the roles does not work.  Also, sometimes this happens: 
Screenshot from 2020-03-27 09-14-45

**Comment by timovdk** [comment]
It throws bad_alloc when roboteam_world is not running

**Comment by timovdk** [comment]
The first tactic in the play also shows the keeper's tactic
Screenshot from 2020-03-27 10-01-57

**Comment by Yuhanun** [comment]
> The first tactic in the play also shows the keeper's tactic
> Screenshot from 2020-03-27 10-01-57

that's wanted behavior for me, would you like me to change it?

**Comment by Yuhanun** [comment]
> It looks nice! However, the scroll bar for the roles does not work. Also, sometimes this happens:
> Screenshot from 2020-03-27 09-14-45

huh?... when does this happen and why?

**Comment by Yuhanun** [comment]
> It throws bad_alloc when roboteam_world is not running

traceback pls?

**Comment by timovdk** [comment]
> > The first tactic in the play also shows the keeper's tactic
> > Screenshot from 2020-03-27 10-01-57
> 
> that's wanted behavior for me, would you like me to change it?

Why is it wanted behavior?

**Comment by timovdk** [comment]
> > It throws bad_alloc when roboteam_world is not running
> 
> traceback pls?

Oh lol, apparently it is the same as Jesse's traceback

Screenshot from 2020-03-27 13-09-06

File: src/interface/widgets/STPVisualizerWidget.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    void STPVisualizerWidget::displayPlay(stp::Play *currentPlay) {
+        updateContent << "Play: ";
+        if (!currentPlay) {
+            updateContent << "None<br>";
+            return;
+        }
+        updateContent << "Some(" << currentPlay->getName() << ")<br>" << tab;
+        std::vector<std::pair<stp::Role*, stp::Status>> states = { currentPlay->getRoleStatuses().begin(), currentPlay->getRoleStatuses().end() };
+        std::sort(states.begin(), states.end(), [](auto const& lhs, auto const& rhs) {
```
Comment (RobotJente) [inline src/interface/widgets/STPVisualizerWidget.cpp:56]: Maybe more descriptive names for the lhs and rhs of the hashmap, but that's minor

**Review by RobotJente — APPROVED** [review]

**Pull request merged**